### ISSUE TITLE
issue modifying disk name to disk label

### ISSUE THREAD

**Issue by rismoney** [terraform-providers/terraform-provider-vsphere#396]
tried upgrading to newest version 1.3.2
in a module i have a vsphere_virtual_machine with a single disk there with this stanza:

```
  disk {
    name = "${var.name}.vmdk"
    thin_provisioned = true
    size = "${var.cdrive}"
  }
```

To make myself forward compliant I tried to rename `name` to label and receive:
* vsphere_virtual_machine.wks: disk label and name cannot be defined at the same time
* module.mybox.vsphere_virtual_machine.wks: 1 error(s) occurred:

**Comment by rismoney** [comment]
I think this is because I had lifecycle set to not destroy. 

This is considered a safe op and won't restart nodes or cause breakage right ?

**Comment by vancluever** [comment]
Hey @rismoney, this is indeed safe - the transition only touches state and does not make a reconfigure request to vSphere, so long as nothing else is changing.

In order for it to be safe though, you **must** make sure the new label of the disk matches the old `name` field. So while it might seem awkward, that would mean that the new label in this case would be the name of the virtual machine, or whatever "${var.name}.vmdk" interpolated to - so something like "foo.vmdk".

Before you apply, review the plan, you should see _only_ the following change:

```
disk.0.name: "foo.vmdk" => ""
disk.0.label: "" => "foo.vmdk"
```

If you see the `<deleted>` or `<remove, keep disk>` annotations, do not proceed.

Let me know how it goes - I'll keep this open until then.

**Comment by rismoney** [comment]
Interestingly I had the disk.0.name  => 'foo/foo.vmdk'
When I changed to label it evaluated to foo.vmdk.
I tried to then adjust variable to the foo/ value, it didn't like it, then went ahead  with just the s/name/label apply and it succeeded.  Subsequent runs were proper noops.  This was on single disk boxes built from clones.

On the multidisk clones I couldn't rename name to label as it seemed to convert disk#s and resulted in <deleted>.  I need to review this week to see.  This impacts even new nodes built on 1.3.2 with label.  So I am not sure if it's my code or a bug. Tbd
